testing.md

```diff
@@ -0,0 +1,180 @@
+
+az keyvault certificate create --vault-name $KEYVAULT_NAME --name <certNameHere> -p "$(az keyvault certificate get-default-policy)"
+```
+
+You can retrieve the value of a Key Vault secret with the following script:
+
+```bash
+# to view a given secret's value
+az keyvault secret show --vault-name $KEYVAULT_NAME --name <secretNameHere> --query value -o tsv
+# to view a given key's value
+az keyvault key show --vault-name $KEYVAULT_NAME --name <yourKeyNameHere> --query "key.n" -o tsv
+# to view a given certificate's value
+az keyvault certificate show --vault-name $KEYVAULT_NAME --name <yourCertificateNameHere> --query cer -o tsv
+```
+### Create a Service Principal
+
+We now need to create a [Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals#service-principal-object) with **Read Only** access to our Azure Key Vault. The Azure Provider will use this Service Principal to access our secrets from our Key Vault.
+
+> You will need to keep track of `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET`, as you will need to place these into the upcoming `secrets.env`.
+
+```bash
+KEYVAULT_RESOURCE_ID=$(az keyvault show -n $KEYVAULT_NAME --query id -o tsv)
+AZURE_CLIENT_ID=$(az ad sp create-for-rbac --name $KEYVAULT_NAME --role Reader --scopes $KEYVAULT_RESOURCE_ID --query appId -o tsv)
+AZURE_CLIENT_SECRET=$(az ad sp credential reset --name $AZURE_CLIENT_ID --credential-description "APClientSecret" --query password -o tsv)
+
+# Assign Read Only Policy for our Key Vault to the Service Principal
+az keyvault set-policy -n $KEYVAULT_NAME --secret-permissions get --spn $AZURE_CLIENT_ID
+az keyvault set-policy -n $KEYVAULT_NAME --key-permissions get --spn $AZURE_CLIENT_ID
+az keyvault set-policy -n $KEYVAULT_NAME --certificate-permissions get --spn $AZURE_CLIENT_ID
+
+```
+The Service Principal(SP) created during this section uses just **Read Only** permissions. This SP is then applied to the Azure Key Vault since we want to limit the Service Principal's credentials to only allow for reading of the keys. This will prevent the chance of manipulating anything on the Key Vault when using the login of this Service Principal.
+
+### Preparing your secrets
+
+Add your secrets to a `secrets.env` file at the application `root` directory.
+
+1. Add all secrets related to the Azure Key Vault, Service Principal, and your Azure Subscription.
+
+> By this point you should have 4 Objects in your Key Vault. 2 Certs, and 2 Objects(Secrets and/or Keys).
+
+    ```bash
+    # secrets.env
+
+    OBJECT1_NAME=<yourKeyVaultSecretName>
+    OBJECT1_TYPE=secret
+    OBJECT1_ALIAS=""
+    OBJECT1_VERSION=""
+
+    OBJECT2_NAME=<yourKeyVaultKeyName>
+    OBJECT2_VALUE=<yourKeyVaultKeyValue>
+    OBJECT2_TYPE=key
+    OBJECT2_ALIAS=<YOUR_KEY_VAULT_KEY_ALIAS>
+    OBJECT2_VERSION=""
+
+    CERT1_NAME=<yourKeyVaultCertName>
+    CERT1_VALUE=<yourKeyVaultCertValue>
```

Comment:
@bdlb77 one small nit: can we write base64 encoded cert value? The test uses base64 encoded